rpcs3/Emu/Cell/Modules/cellGem.cpp

```diff
@@ -819,7 +830,7 @@ error_code cellGemInit(vm::cptr<CellGemAttribute> attribute)
 
 	const auto gem = g_fxo->get<gem_config>();
 
-	if (!attribute || !attribute->spurs_addr || attribute->max_connect > CELL_GEM_MAX_NUM)
+	if (!attribute || !attribute->spurs_addr || attribute->max_connect - 1 > CELL_GEM_MAX_NUM - 1)
```

Comment:
i know this is how the assembly does it.
But shouldn't we use !max_connect for readability and to prevent warnings?

Reply:
you are welcome to change it.
Although it is valid for unsigned overflow to occur and compilers do not optimize out the -1, nor emit a warning.

Reply:
I meant that you are welcome to force push the change.